### PR TITLE
Add README to Prow Directory

### PR DESCRIPTION
**Pull request by chases2**
Copied from another Prow repository with a similar bumping strategy

### CODE REVIEW

File: prow/README.md [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+# and if you are oncall, do the following:
+go get -u k8s.io/test-infra/prow/cmd/mkpj
+
+mkpj --job=FOO > ~/foo.yaml # and answer interactive questions
+
+# Contact #oncall on slack to ensure you are approved to do the following:
+kubectl --context=istio-prow create -f ~/foo.yaml
+```
+
+### Update Config File
```
Comment (clarketm) [inline prow/README.md]: Remove. This doesn't apply.

File: prow/README.md [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+├── LICENSE
+└── README.md
+```
+
+Remember that when you add a job file, you need to set the execution bit!
+```bash
+$ chmod +x prow/my-presubmit.sh
+```
+
+### Test-Infra Prow Jobs
```
Comment (clarketm) [inline prow/README.md]: Remove. This doesn't apply.

File: prow/README.md [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+Test harness will checkout code to directory `$GOPATH/src/github.com/istio/istio`
+
+But in most of our tests we create a soft link between `$GOPATH/src/github.com/istio/<repo>` and `$GOPATH/src/istio.io/<repo>` since we import from `istio.io`
+
+So `_artifacts` dir will be generated before running tests and accessable at both `$GOPATH/src/github.com/istio/istio/_artifacts` and `${GOPATH}/src/istio.io/istio/_artifacts`
+
+Then you can access to the artifact through "artifacts" at [gubernator](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/1025/e2e-suite-rbac-no_auth/1006/)
+
+
+### Prow Bazel Build Image
```
Comment (clarketm) [inline prow/README.md]: Remove. This doesn't apply.

File: prow/README.md [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+      containers:
+      - image: gcr.io/istio-testing/prowbazel:0.4.11
+    # ...
+
+# in plugins.yaml
+my-repo: # ADD THIS BLOCK
+- trigger
+```
+
+### Upload artifacts after job finishes
```
Comment (clarketm) [inline prow/README.md]: Remove. This doesn't apply.

**Comment by chases2** [comment]
Incorrect documentation can only lead to problems; removed the irrelevant sections from README. The example in `Github Trigger` still references istio, but as an example that still applies to this configuration also.

I think it's worth retaining everything in this new version, but I'm most attached to the manual information in `Upgrading Prow`, which is now unique to this Prow instance.

Also fixed a broken link in the top-level README pointing to config.yaml

**Comment by clarketm** [comment]
@chases2 - I agree. Thanks for the documentation and cleanup!

**Comment by krzyzacy** [comment]
@clarketm let me add you to collaborators

**Comment by clarketm** [comment]
Great! Thanks @krzyzacy.

**Comment by krzyzacy** [comment]
lol am I drunk? I cannot find the access to repo settings anymore?

**Comment by clarketm** [comment]
@krzyzacy
image

**Comment by krzyzacy** [comment]
I mean... I obviously don't have that access anymore

**Comment by krzyzacy** [comment]
punt to @fejta

**Comment by clarketm** [comment]
Oh, I misunderstood. Now, I see what you mean. I can ask @fejta.

**Pull request merged**